lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.ZoneHVACComponent.rb

```diff
@@ -0,0 +1,106 @@
+                  zone_hvac_component.to_ZoneHVACPackagedTerminalAirConditioner.get
+                elsif zone_hvac_component.to_ZoneHVACPackagedTerminalHeatPump.is_initialized
+                  zone_hvac_component.to_ZoneHVACPackagedTerminalHeatPump.get
+                elsif zone_hvac_component.to_ZoneHVACTerminalUnitVariableRefrigerantFlow.is_initialized
+                  zone_hvac_component.to_ZoneHVACTerminalUnitVariableRefrigerantFlow.get
+                elsif zone_hvac_component.to_ZoneHVACWaterToAirHeatPump.is_initialized
+                  zone_hvac_component.to_ZoneHVACWaterToAirHeatPump.get
+                elsif zone_hvac_component.to_ZoneHVACEnergyRecoveryVentilator.is_initialized
+                  zone_hvac_component.to_ZoneHVACEnergyRecoveryVentilator.get
+                end
+
+    # Do nothing for other types of zone HVAC equipment
+    return false if zone_hvac.nil?
+
+    # Do nothing if zone hav component isn't assigned to thermal zone
+    return false unless zone_hvac.thermalZone.is_initialized
+
+    # Get baseline system type
+    system_type = zone_hvac.thermalZone.get.additionalProperties.getFeatureAsString('baseline_system_type').get
+
+    # Get non-mechanically cooled flag
+    if zone_hvac.thermalZone.get.additionalProperties.hasFeature('non_mechanically_cooled')
+      nmc_flag = zone_hvac.thermalZone.additionalProperties.hasFeature('non_mechanically_cooled')
+      else nmc_flag = false
+    end
+
+    # Determine the W/cfm
+    fan_efficacy_w_per_cfm = zone_hvac_component_fan_efficacy(system_type, nmc_flag)
+
+    # Convert efficacy to metric
+    fan_efficacy_w_per_m3_per_s = OpenStudio.convert(fan_efficacy_w_per_cfm, 'm^3/s', 'cfm').get
+
+    # Get the fan
+    fan = if zone_hvac.supplyAirFan.to_FanConstantVolume.is_initialized
+            zone_hvac.supplyAirFan.to_FanConstantVolume.get
+          elsif zone_hvac.supplyAirFan.to_FanVariableVolume.is_initialized
+            zone_hvac.supplyAirFan.to_FanVariableVolume.get
+          elsif zone_hvac.supplyAirFan.to_FanOnOff.is_initialized
+            zone_hvac.supplyAirFan.to_FanOnOff.get
+          end
+
+    # Get the maximum flow rate through the fan
+    max_air_flow_rate = nil
+    if fan.autosizedMaximumFlowRate.is_initialized
+      max_air_flow_rate = fan.autosizedMaximumFlowRate.get
+    elsif fan.maximumFlowRate.is_initialized
+      max_air_flow_rate = fan.maximumFlowRate.get
+    end
+    max_air_flow_rate_cfm = OpenStudio.convert(max_air_flow_rate, 'm^3/s', 'ft^3/min').get
+
+    # Set the impeller efficiency
+    fan_change_impeller_efficiency(fan, fan_baseline_impeller_efficiency(fan))
+
+    # Set the motor efficiency, preserving the impeller efficency.
+    # For zone HVAC fans, a bhp lookup of 0.5bhp is always used because
+    # they are assumed to represent a series of small fans in reality.
+    fan_apply_standard_minimum_motor_efficiency(fan, fan_brake_horsepower(fan))
```

Comment:
It looks like fan_brake_horsepower() calls fan_fanpower(), which depends on fan.pressureRise. Since final pressureRise has not yet been set, has it been initialized to a reasonable value at this point?

Reply:
It uses the pressure rise from the `ashrae_90_1_prm.fans.json` file.